cats/src/main/scala/frameless/cats/implicits.scala

```diff
@@ -1,21 +1,45 @@
 package frameless
 package cats
 
-import _root_.cats.implicits._
 import _root_.cats._
+import _root_.cats.kernel.{CommutativeMonoid, CommutativeSemigroup}
+import _root_.cats.implicits._
+import alleycats.Empty
 
 import scala.reflect.ClassTag
 import org.apache.spark.rdd.RDD
 
 object implicits extends FramelessSyntax with SparkDelayInstances {
   implicit class rddOps[A: ClassTag](lhs: RDD[A]) {
-    def csum(implicit m: Monoid[A]): A = lhs.reduce(_ |+| _)
-    def cmin(implicit o: Order[A]): A = lhs.reduce(_ min _)
-    def cmax(implicit o: Order[A]): A = lhs.reduce(_ max _)
+    def csum(implicit m: CommutativeMonoid[A]): A =
+      lhs.fold(m.empty)(_ |+| _)
+    def csumOption(implicit m: CommutativeSemigroup[A]): Option[A] =
+      lhs.aggregate[Option[A]](None)(
+        (acc, a) => Some(acc.fold(a)(_ |+| a)),
+        (l, r) => l.fold(r)(x => r.map(_ |+| x) <+> Some(x))
```

Comment:
I may be reading this wrong, but this seems incorrect to me. Isn't `x` getting added in twice in the case where both `l` and `r` are `Some`?

If so, it's concerning that unit tests didn't catch this. Maybe some places that are calling `toRDD` on a `List` should be calling `parallelize`?

Reply:
I should probably replace the `<+>` operator by `orElse` since I'm basically relying on the fact that the implementation of `<+>` for `Option` is that one.

In essence, I just want to express the `Alternative` between the two `Option` (`<|>`) but in cats this is implemented based on `MonoidK`, and therefore `<|>` becomes `<+>`...

Reply:
Ah right that makes sense. I think that I'm one of the people who advocated for doing this in Cats, so I probably should have realized what was going on here :P

It may be more straightforward to just use `orElse`, but if it works the right way then my main concern is gone. Thanks for the explanation!